src/main/kotlin/org/rust/ide/utils/CallInfo.kt

```diff
@@ -5,44 +5,105 @@
+        fun renderType(): String {
+            return if (type != null && type !is TyUnknown) {
+                type.render(
+                    includeLifetimeArguments = true,
+                    useAliasNames = true,
+                    skipUnchangedDefaultTypeArguments = true
+                )
+            } else {
+                typeRef?.substAndGetText(emptySubstitution) ?: "_"
+            }
+        }
+    }
 
     companion object {
         fun resolve(call: RsCallExpr): CallInfo? {
             val fn = (call.expr as? RsPathExpr)?.path?.reference?.resolve() ?: return null
-            if (fn is RsFunction) return CallInfo(fn)
+            val ty = call.expr.type as? TyFunction ?: return null
 
-            val ty = call.expr.type
-            if (ty is TyFunction) {
-                val parameterNames = getParameterNames(fn) ?: List(ty.paramTypes.size) { "_" }
-                return CallInfo(ty, parameterNames)
+            return when (fn) {
+                is RsFunction -> buildFunctionParameters(fn, ty)
+                else -> buildFunctionLike(fn, ty)
             }
-            return null
         }
 
-        private fun getParameterNames(fn: RsElement): List<String>? {
-            val variant = fn as? RsEnumVariant ?: return null
-            return variant.positionalFields.map { (it.typeReference.type as? TyTypeParameter)?.name ?: "_" }
+        private fun buildFunctionLike(fn: RsElement, ty: TyFunction): CallInfo? {
+            val parameters = getFunctionLikeParameters(fn) ?: return null
+            return CallInfo(buildParameters(ty.paramTypes, parameters))
+        }
+
+        private fun getFunctionLikeParameters(element: RsElement): List<Pair<String?, RsTypeReference?>>? {
+            return when {
+                element is RsEnumVariant -> element.positionalFields.map { null to it.typeReference }
+                element is RsStructItem && element.isTupleStruct ->
+                    element.tupleFields?.tupleFieldDeclList?.map { null to it.typeReference }
+                element is RsPatBinding -> {
+                    val decl = element.parentOfType<RsLetDecl>() ?: return null
+                    val lambda = decl.expr as? RsLambdaExpr ?: return null
+                    lambda.valueParameters.map { (it.patText ?: "_") to it.typeReference }
+                }
+                else -> null
+            }
         }
 
         fun resolve(methodCall: RsMethodCall): CallInfo? {
-            return (methodCall.reference.resolve() as? RsFunction)?.let { CallInfo(it) }
+            val function = (methodCall.reference.resolve() as? RsFunction) ?: return null
+            val type = methodCall.inference?.getResolvedMethodType(methodCall) ?: return null
+            return buildFunctionParameters(function, type)
+        }
+
+        private fun buildFunctionParameters(function: RsFunction, ty: TyFunction): CallInfo {
+            val types = let {
```

Comment:
[minor] Executing a block of statements is usually implemented using `run`
```suggestion
            val types = run {
```